### PR TITLE
Prune metyrics: regression 9/n

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

prune remaining regression metrics
follow-up #6505

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/metrics/regression/r2score.py [hunk 1]
```diff
@@ -98,50 +31,9 @@ def __init__(
...
-        self.sum_squared_error += sum_squared_error
-        self.sum_error += sum_error
-        self.residual += residual
-        self.total += total
-
-    def compute(self) -> torch.Tensor:
-        """
-        Computes r2 score over the metric states.
+        .. deprecated::
+            Use :class:`~torchmetrics.Accuracy`. Will be removed in v1.5.0.
```
Comment (carmocca) [inline pytorch_lightning/metrics/regression/r2score.py:38]: accuracy?

File: pytorch_lightning/metrics/regression/ssim.py [hunk 2]
```diff
@@ -62,44 +33,9 @@ def __init__(
...
-        """
-        preds, target = _ssim_update(preds, target)
-        self.y_pred.append(preds)
-        self.y.append(target)
-
-    def compute(self):
-        """
-        Computes explained variance over state.
+        .. deprecated::
+            Use :class:`~torchmetrics.Accuracy`. Will be removed in v1.5.0.
```
Comment (carmocca) [inline pytorch_lightning/metrics/regression/ssim.py:40]: same

File: pytorch_lightning/metrics/regression/r2score.py [hunk 3]
```diff
@@ -98,50 +31,9 @@ def __init__(
...
-    def update(self, preds: torch.Tensor, target: torch.Tensor):
-        """
-        Update state with predictions and targets.
-
-        Args:
-            preds: Predictions from model
-            target: Ground truth values
         """
-        sum_squared_error, sum_error, residual, total = _r2score_update(preds, target)
+        This implementation refers to :class:`~torchmetrics.Accuracy`.
```
Comment (Borda) [inline pytorch_lightning/metrics/regression/r2score.py:35]:
```suggestion
        This implementation refers to :class:`~torchmetrics.R2Score`.
```

File: pytorch_lightning/metrics/regression/r2score.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/metrics/regression/r2score.py:38]: good chatch

File: pytorch_lightning/metrics/regression/ssim.py [hunk 4]
```diff
@@ -62,44 +33,9 @@ def __init__(
...
-        self.sigma = sigma
-        self.data_range = data_range
-        self.k1 = k1
-        self.k2 = k2
-        self.reduction = reduction
-
-    def update(self, preds: torch.Tensor, target: torch.Tensor):
         """
-        Update state with predictions and targets.
+        This implementation refers to :class:`~torchmetrics.Accuracy`.
```
Comment (Borda) [inline pytorch_lightning/metrics/regression/ssim.py:37]:
```suggestion
        This implementation refers to :class:`~torchmetrics.SSIM`.
```

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**